### ISSUE TITLE
On newer SymPy, sometimes False comes back as 0

### ISSUE THREAD

**Issue by cbm755** [cbm755/octsympy#1084]
At least on SymPy 1.10.1, 

```
octave:26> syms x
octave:27> e = (x == 3) & (x^2 == 9)
e = (sym)
           2    
  x = 3 ∧ x  = 9
octave:28> subs(e, x, 3)
ans = (sym) True
octave:29> subs(e, x, -3)
ans = (sym) False
octave:30> subs(e, x, [3 -3])
ans = (sym) [True  0]  (1×2 matrix)
octave:31> subs(e, x, [-3 3])
ans = (sym) [0  True]  (1×2 matrix)
```
Those zeros should be `False`.  I guess this is a problem with putting non-Expr in Matrices, like many of our later sympy issues.

**Comment by cbm755** [comment]
Happens on SymPy 1.10.1, 1.9.1, 1.8.  Seems ok on <= 1.7.1.  That is before the `setitem` in `polys/matrices/sdm.py`...

Likely that `if value:` again.

**Comment by cbm755** [comment]
Actually not sure we have all cases fixed, reopening until #1107.
